int32 num
---
bool successful
